app/src/apdu_handler.c

```diff
@@ -186,10 +197,61 @@ void handleApdu(volatile uint32_t *flags, volatile uint32_t *tx, uint32_t rx) {
 
                 case INS_SIGN: {
                     CHECK_PIN_VALIDATED()
+                    set_tx_type(tx_type_json);
+                    handleSign(flags, tx, rx);
+                    break;
+                }
+
+                case INS_SIGN_HASH: {
+                    CHECK_PIN_VALIDATED()
+                    set_tx_type(tx_type_hash);
+                    handleSign(flags, tx, rx);
+                    break;
+                }
+
+                case INS_SIGN_TRANSFER: {
+                    CHECK_PIN_VALIDATED()
+                    set_tx_type(tx_type_transfer);
                     handleSign(flags, tx, rx);
                     break;
                 }
 
+                case BCOMP_GET_VERSION: {
+                    CHECK_PIN_VALIDATED()
+                    legacy_handleGetVersion(tx);
+                    break;
+                }
+
+                case BCOMP_VERIFY_ADDRESS: {
+                    CHECK_PIN_VALIDATED()
+                    legacy_handleGetAddr(flags, tx, rx, 1);
```

Comment:
What does 0 and 1 means?

Reply:
Yep. it would be great if we make this a define somewhere.